src/phoebus_driver.cpp

```diff
@@ -398,46 +367,7 @@ TaskCollection PhoebusDriver::RungeKuttaStage(const int stage) {
     if (monopole_gr_active) {
       auto interp_to_monopole =
           tl.AddTask(none, MonopoleGR::InterpolateMatterTo1D<MDT>, sc0.get());
-      sync_region.AddRegionalDependencies(reg_dep_id++, ib, interp_to_monopole);
-    }
-
-    // fluxes now sent/received in mesh-data.
-    auto send_flux =
-        tl.AddTask(none, parthenon::cell_centered_bvars::LoadAndSendFluxCorrections, sc0);
-    auto recv_flux =
-        tl.AddTask(none, parthenon::cell_centered_bvars::ReceiveFluxCorrections, sc0);
-    auto set_flux =
-        tl.AddTask(recv_flux, parthenon::cell_centered_bvars::SetFluxCorrections, sc0);
-
-    // compute the divergence of fluxes of conserved variables
-    auto flux_div =
-        tl.AddTask(set_flux, parthenon::Update::FluxDivergence<MeshData<Real>>, sc0.get(),
-                   dudt.get());
-
-#if SET_FLUX_SRC_DIAGS
-    auto copy_flux_div =
-        tl.AddTask(flux_div, fluid::CopyFluxDivergence<MeshData<Real>>, dudt.get());
-#endif
-
-    auto add_rhs = tl.AddTask(flux_div, SumData<std::string, MeshData<Real>>, src_names,
-                              dudt.get(), gsrc.get(), dudt.get());
-
-    // update step
-    auto avg_data = tl.AddTask(add_rhs, AverageIndependentData<MeshData<Real>>, sc0.get(),
-                               base.get(), beta);
-    auto update = tl.AddTask(avg_data, UpdateIndependentData<MeshData<Real>>, sc0.get(),
-                             dudt.get(), beta * dt, sc1.get());
-
-    if (rad_mocmc_active) {
-      auto impl_update = tl.AddTask(update, radiation::MOCMCFluidSource<MeshData<Real>>,
-                                    sc1.get(), beta * dt, fluid_active);
-      auto impl_edd =
-          tl.AddTask(impl_update, radiation::MOCMCEddington<MeshData<Real>>, sc1.get());
-      update = impl_update | update;
-    } else if (rad_moments_active) {
-      auto impl_update = tl.AddTask(update, radiation::MomentFluidSource<MeshData<Real>>,
-                                    sc1.get(), beta * dt, fluid_active);
-      update = impl_update | update;
+      sync_region_6.AddRegionalDependencies(reg_dep_id++, ib, interp_to_monopole);
```

Comment:
it's hard to see what's going on here from the diff... it's ideal to minimize the number of regions, since that minimizes fences. Are you doing that?

Reply:
I should actually take another pass at this now that the "right" algorithm is a bit more clear which should inform what needs to be communicated when. I'll update here when Ive done that.

I think ultimately this will need to be retooled a bit anyway once all our tasks are available on a per-`MeshData` basis (for example, particles).